services/coincap/coincap-changepercent24hr.tester.js

```diff
@@ -0,0 +1,43 @@
+import { isPercentage } from '../test-validators.js'
+import { createServiceTester } from '../tester.js'
+export const t = await createServiceTester()
+
+t.create('request for existing asset with positive')
+  .get('/bitcoin.json')
+  .intercept(nock =>
+    nock('https://api.coincap.io')
+      .get('/v2/assets/bitcoin')
+      .reply(200, {
+        data: { changePercent24Hr: '1.4767080598737783', name: 'Bitcoin' },
+      })
+  )
+  .expectBadge({
+    label: 'Bitcoin',
+    message: '1.00%',
+    color: 'brightgreen',
+  })
+
+t.create('request for existing asset with negative')
+  .get('/bitcoin.json')
+  .intercept(nock =>
+    nock('https://api.coincap.io')
+      .get('/v2/assets/bitcoin')
+      .reply(200, {
+        data: { changePercent24Hr: '-1.4767080598737783', name: 'Bitcoin' },
+      })
+  )
+  .expectBadge({
+    label: 'Bitcoin',
+    message: '-1.00%',
+    color: 'red',
+  })
+
+t.create('change percent 24hr').get('/bitcoin.json').expectBadge({
+  label: 'Bitcoin',
+  message: isPercentage,
```

Comment:
Looking at the build for this branch it is failing with
`ValidationError: message mismatch: "value" does not match any of the allowed types`
because at the time the tests ran the change was negative and the validator isn't expecting a leading `-`.
We'll need to adjust the validator to allow negative values.

Reply:
I have added negative possibilities on `test-validators.js` ok?